autoendpoint/src/error.rs

```diff
@@ -152,6 +152,11 @@ impl ApiErrorKind {
         }
     }
 
+    /// Should the User record be removed?
+    pub fn user_still_valid(&self) -> bool {
+        !matches!(self, ApiErrorKind::InvalidToken)
```

Comment:
`InvalidToken` is only produced by the extractors to signal the passed in token didn't parse. I think the only case we now should cover here is `RouterError::NotFound` but it's possible we don't need any of this at all (see my other comment)